libraries/CurieBLE/src/BLEDevice.h

```diff
@@ -361,83 +361,70 @@ class BLEDevice
     //void scanForAddress(String address); // Not include in baseline. Add here as feature for feature release.
     
     /**
-     * @brief   Start scanning for peripherals without filter
+     * @brief   Start scanning for peripherals with the option of accepting all detectable
+     *          Peripherals or just the newly detected.
      *
-     * @param   none
-     *
-     * @return  none
-     *
-     * @note  none
-     */
-    void scan();
-    
-    /**
-     * @brief   Start scanning for peripherals with filter
-     *
-     * @param[in] withDuplicates        true - with duplicate filter
-     *                                  false- without duplicate filter
-     *
-     * @return  none
-     *
-     * @note  option to filter out duplicate addresses for Arduino.
-     *          The current only support fileter duplicate mode.
-     */
-    void scan(bool withDuplicates);
-    
-    /**
-     * @brief   Start scanning for peripherals and filter by device name in ADV
-     *
-     * @param   name    The device's local name.
+     * @param[in] withDuplicates        true - return all detectable Peripherals.
+     *                                  false- return only new Peripherals.
      *
      * @return  none
      *
-     * @note   option to filter out duplicate addresses for Arduino.
-     *          The current only support fileter duplicate mode.
+     * @note  When, withDuplicates = true (default), accept all detectable Peripherals.
+     *        No Peripheral filtering process applied to the scan result.
      */
-    void scanForName(String name);
+    void scan(bool withDuplicates = true);
```

Comment:
@SidLeung could we please change the `withDuplicates` default arg to false here.

Remember the overall goal is to have something similar to the Apple CoreBluetooth API: https://developer.apple.com/reference/corebluetooth/cbcentralmanager/1518986-scanforperipherals

which by default doesn't allow duplicates: https://developer.apple.com/reference/corebluetooth/cbcentralmanagerscanoptionallowduplicateskey